### DISCUSSION TITLE
How to query two components simultaneously?

### DISCUSSION THREAD

**Discussion by ArikRahman** [bevyengine/bevy#6435] [Q&A] [answered]
The tutorial shows how to query one component (the name):

`
fn greet_people(query: Query<&Name, &Age, With<Person>>) {
    for name in query.iter() {
        println!("hello {}!", name.0);
    }
}
`

How do I query two components, say the age and the name of a person?

**Comment by rparrett** [comment, marked as answer]
[`Query`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html) takes two generic parameters, a query and a filter.

To query for multiple components or to query with multiple filters, you can use a tuple in either position.

```rust
fn greet_people(
    query: Query<(&Name, &Age), (With<Person>, With<SillyHat>)>
) {
    for (name, transform) in query.iter() {
        /* ... */
    }
}
```

**Comment by harudagondi** [comment]
```rust
fn greet_people(query: Query<(&Name, &Age), With<Person>>) {
	for (name, age) in query.iter() {
		println!("Hello {}! You are {} years old.", name.0, age.0); 
	} 
}
```

You use a tuple.

**Comment by ArikRahman** [comment]
Thanks guys, I played around with it and the tuple method worked!
![image](https://user-images.githubusercontent.com/40479733/199146757-78c0a6d8-7e7c-459a-8878-27646d959731.png)

**Reply by rparrett** [reply]
`With<Person>` is the filter part of the query and should be the second generic parameter of `Query`.

```rust
fn greet_people(
    query: Query<(&Name, &Gender), With<Person>>
) {
    for (name, gender) in query.iter() {
        /* ... */
    }
}
